protocols/raft/src/main/java/io/atomix/protocols/raft/roles/AbstractAppender.java

```diff
@@ -457,10 +457,10 @@ protected InstallRequest buildInstallRequest(
         final DefaultRaftMember leader = raft.getLeader();
         request =
             InstallRequest.builder()
-                .withTerm(raft.getTerm())
+                .withCurrentTerm(raft.getTerm())
                 .withLeader(leader != null ? leader.memberId() : null)
                 .withIndex(snapshot.index())
-                .withSnapshotTerm(snapshot.term())
```

Comment:
Can these terms really be different?

Reply:
Yes, we send the term at the time of the request as a check (see PassiveRole onInstall), but we also need the term of the entry the snapshot refers to to know if we need to discard uncommitted entries (also in uninstall in the PassiveRole), and these can be different

Reply:
thanks